src/components/Identity/createComponent.js

```diff
@@ -12,13 +12,10 @@ export default ({
   let ecid;
   return {
     lifecycle: {
-      // TODO: It would probably be best to query on the data collection payload level
-      // rather than the event. It seems like a payload-level thing and would save
-      // space whenever we start supporting multiple events per payload.
-      onBeforeEvent({ event }) {
-        addEcidQueryToEvent(event);
-      },
       onBeforeRequest({ payload, onResponse }) {
+        // Querying the ECID at the payload level so would return it on all requests,
+        // like `setConsent`.
```

Comment:
I think you missed a word in this comment.  We should add in the comment why we need to get it:
1. to set the legacy ecid cookie.
2. to make it available for the getIdentity command.

And also add in the comment that the ecid can change in response to changes in the identityMap. (i.e. when a user logs in)